### PR TITLE
Fix/Issue 162 Infinite loop with non-editable block inside toggle

### LINKED ISSUE #162: Infinite loop with non-editable block inside toggle
### Precheck

- [x] Do a quick search and make sure the bug has not been reported yet.
- [x] Please disclose security vulnerabilities privately at [oss@kommit.co](mailto:oss@kommit.co).
- [x] Finally, be nice and have fun!.

### Description

Pressing shift+tab to remove a block from inside a toggle will result in an infinite loop when a non-editable block (e.g. [delimiter](https://github.com/editor-js/delimiter)) is present inside the toggle.

### Environment

* Editor.js version: 2.25.0

### Current behavior

See https://codesandbox.io/s/editorjs-toggle-block-test-case-bz2e4r for reproducible steps.

### Expected behavior

In the above case, we should expect to see the caret moved to the first paragraph with the toggle set to having 2 items.

### Additional context

Seems the offending line is https://github.com/kommitters/editorjs-toggle-block/blob/v0.3.2/src/index.js#L367; there's no check that the index hasn't changed inside the loop.

### PR DESCRIPTION
**Pull request by karolrojas07**
Closes #162 

# Solution

### CODE REVIEW

**Review by miguelnietoa — CHANGES_REQUESTED** [review]
Hello @karolrojas07, thanks a lot for solving the issue!! 😄 💯 

Some changes are required ⬇️

File: src/index.js [hunk 1]
```diff
@@ -87,6 +87,31 @@ export default class ToggleBlock {
...
+  isAToggleRoot(holder) {
+    const toggle = holder.querySelector('.toggle-block__selector');
+    if (typeof toggle !== 'undefined' && toggle !== null) {
+      return true;
+    }
+    if (holder.classList.contains('toggle-block__selector')) {
+      return true;
+    }
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (karolrojas07) [reply src/index.js:113]: Awesome!, I didn't know I can check if the object exists using Boolean class. This is so much better and compact. Thank you.

File: src/index.js [hunk 2]
```diff
@@ -351,39 +376,51 @@ export default class ToggleBlock {
     }
   }
 
+  /**
+  * Returns the toggle's root index, given the index of one of its children
+  *
+  * @param {number} entryIndex - block index
+  * @returns {number} The Toggle's root index
+  */
```
[the remark this answers is not in the document]
Comment (karolrojas07) [reply src/index.js:384]: Done!

File: test/testHelpers.js [hunk 3]
```diff
@@ -30,20 +46,54 @@ export function createNestedBlock(toggleBlock, data) {
...
+
+  children.forEach((child, idx) => {
+    if (idx === 0) { child.setAttribute('status', status); }
+    if (status === 'closed') {
+      child.setAttribute('hidden', true);
+    } else {
+      child.setAttribute('hidden', false);
+    }
+  });
+}
```
[the remark this answers is not in the document]
Comment (karolrojas07) [reply test/testHelpers.js:95]: This has been refactored. Thank you!

File: test/testHelpers.js [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+  const codexEditor = document.createElement('div');
+  codexEditor.classList.add('codex-editor');
+  codexEditor.appendChild(editorRedactor);
+
+  const editorJS = document.createElement('div');
+  editorJS.setAttribute('id', 'editorjs');
+  editorJS.appendChild(codexEditor);
+
+  document.body.appendChild(editorJS);
+}
```
[the remark this answers is not in the document]
Comment (karolrojas07) [reply test/testHelpers.js:30]: Done!

**Comment by miguelnietoa** [comment]
Thank you for submitting the changes! @karolrojas07 
In order to merge this change we need you to sign the commit. See https://docs.github.com/es/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by miguelnietoa** [comment]
@karolrojas07 only the bundle.js file needs to be updated by running the command `npm run build`

**Review by miguelnietoa — APPROVED** [review]
🚀

**Pull request merged**